packages/ibm-products/src/components/APIKeyModal/APIKeyModal.tsx

```diff
@@ -109,13 +119,46 @@ export let APIKeyModal: React.FC<APIKeyModalProps> = forwardRef(
       ref: copyRef,
     };
     const blockClass = `${pkg.prefix}--apikey-modal`;
+    const localRef = useRef(undefined);
+    const modalRef = (ref || localRef) as MutableRefObject<HTMLDivElement>;
+    const { firstElement, keyDownListener } = useFocus(
+      modalRef,
+      selectorPrimaryFocus
+    );
 
     useEffect(() => {
       if (copyRef.current && open && apiKeyLoaded) {
         copyRef.current.focus();
       }
     }, [open, apiKeyLoaded]);
 
+    useEffect(() => {
```

Comment:
Since we now have a very similar `useEffect` here and in `TearsheetShell`, we should think about putting this inside a custom hook that is used in both components. I don't want to hold up this PR longer, so lets look at that after we get this merged.

Reply:
@matthewgallo Sounds good, may be I can create a refactor ticket for that and raise PR.